comet/utils/db_manager.py

```
@@ -0,0 +1,614 @@
+                        self._lock_retry_count -= 1
+                    return len(batch_data)
+
+            except Exception as e:
+                error_msg = str(e).lower()
+
+                if "locked" in error_msg:
+                    had_lock_error = True
+                    self._lock_retry_count += 1
+
+                    if attempt < max_retries:
+                        wait_time = min(16, (2**attempt)) + random.uniform(0.1, 0.5)
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)